src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -7963,8 +7984,9 @@ internal string GetDebuggerDisplay()
                 var pooledBuilder = PooledStringBuilder.GetInstance();
                 var builder = pooledBuilder.Builder;
                 builder.Append(" ");
-                for (int i = this.Capacity - 1; i >= 0; i--)
-                    builder.Append(_state[i] ? '?' : '!');
+                int n = Math.Min(Capacity, 8);
```

Comment:
>Math.Min(Capacity, 8) [](start = 24, length = 21)

I don't get this at all.  Won't this result in an index out of bounds?

Reply:
The intent is to limit the number of bits displayed in the debugger to at most 8.

---
In reply to: [347695009](https://github.com/dotnet/roslyn/pull/39778#discussion_r347695009) [](ancestors = 347695009)